### DISCUSSION TITLE
any suggestion for audio only mode?

### DISCUSSION THREAD

**Discussion by mdehrlich** [GerryDazoo/Slinger#195] [General]
Sometimes I am outside walking or biking, and my connection is poor.  On slingplayer, it has an audio only "resolution".  Any suggestion how I might do this using slinger?

**Comment by RandomMan719** [comment]
![37B209BC-A994-410F-9C72-02DACB999170](https://user-images.githubusercontent.com/116921195/198751811-42f20159-19bc-41cc-8490-882e6bbd07da.jpeg)
In the sbcore.ini file there is a setting for “AudioOnly=0”
I don’t know if Slinger supports that option or not.
AudioOnly=1 should activate it, you could try putting that in your config.ini and see if it does anything.

**Reply by mdehrlich** [reply]
In slinger, it appears that the two bytes following VideoSmoothness are 3 and 1 followed by  AudioBitRate.  This matches your template, but not the values.   0 might enable audio only?

**Reply by GerryDazoo** [reply]
I didn't know about that  option, And yes it might be buried in the setup video message but the perl code I used for a template just had the bits hard coded so that's what I did. 
So no AudioOny will not work. I'm faling behind here before the 9th. Could either of you gentlemen get a tcpdump/wireshark trace of a session startup with and without AudioOnly set so we cab figure out which bit to toggle.
In the meantime, may I suggest just minimizing the video bandwidth for your cellular connection. I have two configs one for high-resolution and one for streaming over cellular. Works great. I think that will get you over the hump.

**Comment by mdehrlich** [comment]
I have tried to get wireshark to record slingplayer packets, but I have a problem.  Maybe I just don't know how to do it. I can record slingplayer if it is running on my laptop, and wireshark is also on my laptop.  The problem is that the windows based slingplayer does not support audio only (AO) mode.  AO only shows up as an option on my android cellphone slingplayer.  But when I run on cellphone, nothing shows up in wireshark.  DO you think I can just try modifying the AO byte in the sling command in your python code?  Also do you know what happens if resolution is set to zero?

**Reply by RandomMan719** [reply]
I was able to set slingplayer 1.5.1 in windows 10 to audio only mode by setting the AudioOnly=0 to AudioOnly=1 in the SbCore.ini file.  Using Notepad with Admin privileges.  When in audio only mode, the screen just had a graphic with music notes or something.  I don't know how to do a wireshark, but I might see if I can figure it out tonight or tomorrow.  In the SbCore.ini it does show 0 resolution = no video, as well

**Comment by mdehrlich** [comment]
Audio only works perfectly by setting resolution to 0.  Checking in VLC codec info, the video stream is completely missing.  Only one audio stream remains.  Garry, you need to remove the block on 0 resolution.  I have an M1 and a 350.  Works on both. Thanks to RandomMan719

**Comment by mdehrlich** [comment]
Gerry, not heard from you yet.  Could you please remove the block on resolution 0?  Thx, mike

**Reply by GerryDazoo** [reply]
Yes, I will do that. I did not know resolution=0 was valid and implied audio only. If you're running the python code it's a simple two character patch. On the latest version change line 828
        if resolution == 0 or resolution > 16 : 
to
        if resolution < 0 or resolution > 16 :

**Comment by GerryDazoo** [comment]
Audio Only is available now in the latest release.
